app/controllers/media_controller.rb

```diff
@@ -21,6 +21,12 @@ def download
         elsif stream_from_s3?
           body = Services.cloud_storage.load(asset)
           send_data(body.read, **AssetManager.content_disposition.options_for(asset))
+        elsif proxy_via_nginx?
+          url = Services.cloud_storage.presigned_url_for(asset)
+          headers['X-Cloud-Storage-URL'] = url
+          headers['X-Cloud-Storage-Host'] = URI.parse(url).host
```

Comment:
I've just noticed that you've explained this in an example name below, so I think it's less important to explain in the commit note.